authLdap.php

```diff
@@ -594,22 +625,28 @@ function authLdap_groupmap($username, $dn)
     // Check whether the user is member of one of the groups that are
     // allowed acces to the blog. If the user is not member of one of
     // The groups throw her out! ;-)
-    // If the user is member of more than one group only the first one
-    // will be taken into account!
-
-    $role = '';
+    $roles = [];
     foreach ($authLDAPGroups as $key => $val) {
         $currentGroup = explode($authLDAPGroupSeparator, $val);
         // Remove whitespaces around the group-ID
         $currentGroup = array_map('trim', $currentGroup);
         if (0 < count(array_intersect($currentGroup, $grp))) {
-            $role = $key;
-            break;
+            $roles[] = $key;
         }
     }
 
-    authLdap_debug("Role from LDAP group: {$role}");
-    return $role;
+    // Default: If the user is member of more than one group only the first one
+    // will be taken into account!
+    // This filter allows you to return multiple user roles. WordPress
+    // supports this functionality, but not natively via UI from Users
+    // overview (you need to use a plugin). However, it's still widely used,
+    // for example, by WooCommerce, etc. Use if you know what you're doing.
+    if (apply_filters('authLdap_allow_multiple_roles', false) === false && count($roles)) {
```

Comment:
Can we use `count($roles) < 1` here instead?

Reply:
Glad you like it, but there should be `count($roles) > 0` but I can change that for sure.